### DISCUSSION TITLE
How to use null with Prisma.join in a raw query?

### DISCUSSION THREAD

**Discussion by andreasvh-conceto** [prisma/prisma#14769] [Q&A] [answered]
Hello,

currently iam facing an issue, that i need records for specific values, but it can happen, that i also need to look for records with null. I want to create a queryRaw, not a queryRawUnsafe, because of sql injection. Ist that possible?:

```
const names = ['Peter', 'Klaus', null, 'Hugo',] 
const result =
  await prisma.$queryRaw`SELECT * FROM User WHERE id IN (${Prisma.join(ids)})`
```

Currently iam facing the issue:
"name" is of type numeric but expression is of type text
    HINT: You will need to rewrite or cast the expression.

Any idea? I tried Prisma.DbNull, but this was not solving the issue. 

Thank you for your help

Best regards
Andreas

**Comment by austincrim** [comment]
Hey there @andreasvh-conceto 👋,

What's the type of the `User.id` field? You might need to perform some kind of casting in the SQL expression. 

What database provider are you using?

**Reply by vladi-stevanovic** [reply]
👋 Hello @andreasvh-conceto I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by andreasvh-conceto** [comment, marked as answer]
Hello,
the userId field is of type bigint and auto genarated. The issue was the type field as you mentioned. Per default null is of type text in postgres. In my raw query i had to use null::bigint  , or null::int to cast the null to the right type. With that in mind and assumning the user has an age (integer) you need to passs null values and cast them to the right type:
So instead of:
` await prisma.$queryRaw`SELECT * FROM User WHERE age IN (null)`
I had to use:
` await prisma.$queryRaw`SELECT * FROM User WHERE age IN (null:int)`

This helped me solving the issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
